gpytorch/priors/torch_priors.py

```diff
@@ -23,6 +24,34 @@ def __init__(self, loc, scale, validate_args=False, transform=None):
         _bufferize_attributes(self, ("loc", "scale"))
         self._transform = transform
 
+    def expand(self, batch_shape):
+        return Normal.expand(self, batch_shape, _instance=self)
```

Comment:
kind of annoying  how you can't call the superclass method here ...

Reply:
Agreed. Unfortunately, `torch.distributions.Distribution` explicitly forbids it. I've no idea why, but it means we need to define `expand` methods for all of our priors that are just a 1 line boilerplate.

The other thing we'll need to add to priors is `rsample` functionality for any priors we want to use for sampling -- e.g., I define a `UniformPrior` in this PR mostly because `SmoothedBoxPrior` doesn't have an rsample yet.

Reply:
> Unfortunately, torch.distributions.Distribution explicitly forbids it. I've no idea why, but it means we need to define expand methods for all of our priors that are just a 1 line boilerplate.

Yeah I noticed that as well. I guess you could make a `LessOpinionatedTorchDistributionMixin` that reimplements [this](https://github.com/pytorch/pytorch/blob/master/torch/distributions/distribution.py#L255-L260) and removes the error there :)